backend/fms_core/viewsets/dataset.py

```diff
@@ -58,42 +58,22 @@ def add_run_processing(self, request, *args, **kwargs):
         else:
             return Response(self.get_serializer(datasets.values(), many=True).data)
 
-    @transaction.atomic
     @action(detail=True, methods=["patch"])
     def set_release_status(self, request, pk):
         data = request.data
-        release_status = data.get("release_status")
-        exceptions = data.get("exceptions")
-        filters = data.get("filters")
+        readset_updates: dict[int, ReleaseStatus.RELEASED | ReleaseStatus.BLOCKED] = data.get("updates", {})
+        filters = data.get("filters", {})
 
-        filtered_readsets = Readset.objects.filter(dataset=pk).exclude(validation_status=ValidationStatus.AVAILABLE)
-        if not filtered_readsets.exists():
-            transaction.set_rollback(True)
-            return HttpResponseBadRequest(f"Run must first be validated before release status can be changed.")
-        if filters:
-            filtered_readsets = filtered_readsets.filter(**filters)
-        try:
-            # set release flag of all files except exceptions
-            included_readsets = filtered_readsets.filter(~Q(id__in=exceptions))
-            for included_readset in included_readsets:
-                included_readset.release_status = release_status
-                included_readset.release_status_timestamp = timezone.now()
-                included_readset.released_by = request.user
-                included_readset.save()
-
-            # set release flag of exceptions to the opposite flag
-            excluded_readsets = Readset.objects.filter(id__in=exceptions)
-            opposite_status = [ReleaseStatus.AVAILABLE, ReleaseStatus.BLOCKED, ReleaseStatus.RELEASED][release_status]
-            for excluded_readset in excluded_readsets:
-                excluded_readset.release_status = opposite_status
-                excluded_readset.release_status_timestamp = timezone.now()
-                excluded_readset.released_by = request.user
```

Comment:
This is just gone ...

Reply:
Is there a problem with that?

Reply:
It's right here ;)
```python
        releease_status_timestamp = timezone.now()
        for readset in readsets:
            readset.release_status = readset_updates[str(readset.id)]
            readset.release_status_timestamp = releease_status_timestamp
            readset.save()
```

Reply:
released_by is not longer set...

Reply:
I'm surprised it let me release readsets despite not setting released_by.

Reply:
but i think it's there already?
```python
        if self.release_status_timestamp is not None and self.released_by is None:
            add_error("released_by", f"The user that changed the release_status need to be recorded.")
        if self.release_status_timestamp is None and self.released_by is not None:
            add_error("released_by", f"Readset release status was never set. Cannot set released_by.")
        if self.validation_status_timestamp is not None and self.validated_by is None:
            add_error("validated_by", f"The user that changed the validation_status need to be recorded.")
        if self.validation_status_timestamp is None and self.validated_by is not None:
            add_error("validated_by", f"Readset validation status was never set. Cannot set validated_by.")
```

Reply:
I did not test it ... as far as i know this might never have worked ...

Reply:
Mmm i think you might be bypassing validation by using the manager save on the manager instead of the readset object... Try adding .all() to your queryset and removeing the released_by... it should return an error.